nxc/protocols/smb.py

```diff
@@ -405,7 +401,6 @@ def kerberos_login(self, domain, username, password="", ntlm_hash="", aesKey="",
 
     def plaintext_login(self, domain, username, password):
         # Re-connect since we logged off
-        self.create_conn_obj()
```

Comment:
we are missing the first create_conn_obj for the first user no ?

Reply:
It should be fine. From my experience the connection was still intact, despite the login try in `enum_host_info()`:
![image](https://github.com/user-attachments/assets/014e4357-3ead-4941-985c-29cb14f71edf)

Reply:
Actually I should test this against a server with null session enabled, just to make sure that doesn't break it

Reply:
from the comment 
> # DCs seem to want us to logoff first, windows workstations sometimes reset the connection

workstation too :)

Reply:
Damn it, non-dcs are not happy with that:
![image](https://github.com/user-attachments/assets/f3f6bf4a-5bd2-4e4d-9184-1c878f0a9196)

Reply:
Added a check in enum_host, so we create a new connection object if we want to authenticate:
![image](https://github.com/user-attachments/assets/f280430b-98b2-4ca8-a5db-18b3b7c42c2f)